### PR TITLE
Server complete

### PR DESCRIPTION
**Pull request by Alexithemia**

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        console.log("[31mPlease include a specifed filename to be saved to, ex: 'savethisthing.html'[39m");
+        process.exit(1);
+      }
+      save = process.argv[i + 1];
+    } else if (process.argv[i] === '-h') {
+      help = true;
+    } else if (process.argv[i] === '-H') {
+      getHeader = true;
+    }
+  } else {
```
Comment (taesup) [inline client.js:52]: So there could be an issue where if someone tried to use an option that this client does support at the end of the command. Then that last option will end up being the url.

File: client.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  POST: true,
+  PUT: true,
+  DELETE: true,
+  HEAD: true,
+  OPTIONS: true
+}
+
+for (let i = 2; i < process.argv.length; i++) {
+  if (options[process.argv[i]]) {
+    if (methods[process.argv[i]]) {
```
Comment (taesup) [inline client.js]: So curl has a flag for setting the method, I believe it's -X, so it's best to re-write this to follow that convention.

File: server.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+const net = require('net');
+const fs = require('fs');
+
+function fetchHtmlFiles(filePath) {
+  return fs.readFileSync(filePath);
```
Comment (taesup) [inline server.js]: Sync based methods are not allowed.

**Review by taesup — CHANGES_REQUESTED** [review]

**Comment by jaygiang** [comment]
LGTM:white_check_mark:

**Pull request merged**